Resources/ProtobufDumper/ProtobufDumper/ProtobufDumper.cs

```diff
@@ -0,0 +1,957 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using google.protobuf;
+using ProtoBuf;
+
+namespace ProtobufDumper
+{
+    class ProtobufDumper
+    {
+        public delegate void ProcessProtobuf( string name, string buffer );
+
+        private readonly IList<FileDescriptorProto> protobufs;
+        private readonly Stack<string> messageNameStack;
+        private readonly Dictionary<string, ProtoNode> protobufMap;
+        private readonly Dictionary<string, ProtoTypeNode> protobufTypeMap;
+
+        private class ProtoNode
+        {
+            public string Name;
+            public FileDescriptorProto Proto;
+            public List<ProtoNode> Dependencies;
+            public HashSet<FileDescriptorProto> AllPublicDependencies;
+            public List<ProtoTypeNode> Types;
+            public bool Defined;
+        }
+
+        private class ProtoTypeNode
+        {
+            public string Name;
+            public FileDescriptorProto Proto;
+            public object Source;
+            public bool Defined;
+        }
+
+        [ProtoContract]
+        private class ExtensionPlaceholder : IExtensible
+        {
+            private global::ProtoBuf.IExtension extensionObject;
+            global::ProtoBuf.IExtension global::ProtoBuf.IExtensible.GetExtensionObject( bool createIfMissing )
+            { return global::ProtoBuf.Extensible.GetExtensionObject( ref extensionObject, createIfMissing ); }
```

Comment:
The formatting of this method is inconsistent with the rest of our already-inconsistent codebase. Either put braces on their own lines or consider making this an Expression-Bodied function.